eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerFindAll.kt

```diff
@@ -0,0 +1,74 @@
+package com.arsiu.eduhub.controller.nats.assignment
+
+import com.arsiu.eduhub.controller.nats.NatsController
+import com.arsiu.eduhub.mapper.AssignmentNatsMapper
+import com.arsiu.eduhub.service.AssignmentService
+import com.arsiu.eduhub.v2.assignmentsvc.NatsSubject.ASSIGNMENT_FIND_ALL
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentProto
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentProto.AssignmentResponse
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentProto.AssignmentResponse.Failure
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentProto.AssignmentResponse.Failure.CustomError
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentProto.AssignmentResponse.Success
+import com.arsiu.eduhub.v2.assignmentsvc.input.reqreply.assignment.FindAllAssignmentRequestProto.FindAllAssignmentRequest
+import com.arsiu.eduhub.v2.assignmentsvc.output.reqreply.assignment.FindAllAssignmentResponseProto.FindAllAssignmentResponse
+import com.google.protobuf.Parser
+import io.nats.client.Connection
+import org.springframework.stereotype.Component
+
+@Component
+class AssignmentNatsControllerFindAll(
+    override val connection: Connection,
+    private val service: AssignmentService,
+    private val mapper: AssignmentNatsMapper
+) : NatsController<FindAllAssignmentRequest, FindAllAssignmentResponse> {
+
+    override val subject: String = ASSIGNMENT_FIND_ALL
+    override val parser: Parser<FindAllAssignmentRequest> = FindAllAssignmentRequest.parser()
+
+    override fun handler(request: FindAllAssignmentRequest): FindAllAssignmentResponse {
+        return try {
```

Comment:
You can also use runCatching like in sample higher

Reply:
`+`